### PR TITLE
[NEAT-162] Fix DMS importer

### PR DESCRIPTION
**Pull request by doctrino**

### CODE REVIEW

File: cognite/neat/utils/cdf_loaders/_data_modeling.py [hunk 1]
```diff
@@ -160,22 +163,26 @@ def _retrieve_view_ancestors(self, parents: list[ViewId], cache: dict[ViewId, Vi
             parents: The parents of the view to retrieve all ancestors for
             cache: The cache to store the views in
         """
-        parent_ids = parents
+        parent_ids = parents.copy()
```
Comment (doctrino) [inline cognite/neat/utils/cdf_loaders/_data_modeling.py:166]: Avoid to mutate input list as it is an undesired side effect.

File: cognite/neat/utils/cdf_loaders/_data_modeling.py [hunk 2]
```diff
@@ -160,22 +163,26 @@ def _retrieve_view_ancestors(self, parents: list[ViewId], cache: dict[ViewId, Vi
...
         found: list[View] = []
+        found_ids: set[ViewId] = set()
         while parent_ids:
-            to_lookup = []
+            to_lookup: set[ViewId] = set()
             grand_parent_ids = []
             for parent in parent_ids:
-                if parent in cache:
+                if parent in found_ids:
+                    continue
```
Comment (doctrino) [inline cognite/neat/utils/cdf_loaders/_data_modeling.py:174]: The current implementation can return the same view multiple times, this fixes that.

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -31,7 +31,7 @@ repos:
   - hooks:
     - id: mypy
       name: mypy
-      entry: dmypy run -- cognite/neat/
+      entry: dmypy run -- cognite/neat
```
Comment (doctrino) [inline .pre-commit-config.yaml:34]:
```suggestion
      entry: dmypy run -- cognite/neat/

```

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -32,6 +32,7 @@ repos:
     - id: mypy
       name: mypy
       entry: dmypy run -- cognite/neat/
+
```
Comment (doctrino) [inline .pre-commit-config.yaml:35]:
```suggestion

```

**Review by nikokaoja — APPROVED** [review]

File: cognite/neat/utils/cdf_loaders/_data_modeling.py [hunk 5]
```diff
@@ -160,22 +163,26 @@ def _retrieve_view_ancestors(self, parents: list[ViewId], cache: dict[ViewId, Vi
...
+                elif parent in cache:
                     found.append(cache[parent])
                     grand_parent_ids.extend(cache[parent].implements or [])
                 else:
-                    to_lookup.append(parent)
+                    to_lookup.add(parent)
```
Comment (nikokaoja) [inline cognite/neat/utils/cdf_loaders/_data_modeling.py:179]: great catch !

**Review by nikokaoja — APPROVED** [review]

**Pull request merged**